lib/bamboo/adapters/mailgun_adapter.ex

```diff
@@ -116,17 +139,50 @@ defmodule Bamboo.MailgunAdapter do
     end)
   end
 
-  @mailgun_message_fields ~w(from to cc bcc subject text html)a
+  defp put_attachments(body, %Email{attachments: []}), do: body
+  defp put_attachments(body, %Email{attachments: attachments}) do
+    attachment_data =
+      attachments
+      |> Enum.reverse
+      |> Enum.map(&prepare_file(&1))
+
+    Map.put(body, :attachments, attachment_data)
+  end
 
-  def filter_non_empty_mailgun_fields(map) do
-    Enum.filter(map, fn({key, value}) ->
-      # Key is a well known mailgun field or is an header or custom var field and its value is not empty
-      (key in @mailgun_message_fields || String.starts_with?(Atom.to_string(key), ["h:", "v:"])) && !(value in [nil, "", []])
+  defp prepare_file(%Attachment{path: nil} = attachment) do
+    {"", attachment.data,
+     {"form-data",
+      [{"name", ~s/"attachment"/},
+       {"filename", ~s/"#{attachment.filename}"/}]},
+     []}
+  end
+  # what is this second case???
```

Comment:
Good catch, I think we can remove it. Or better to remove the first one that matches for `%Attachment{path: nil}` because we don't care about it's value